### ISSUE TITLE
OioEventLoop is not compatible with OioSctpChannel/OioSctpServerChannel

### ISSUE THREAD

**Issue by jestan** [netty/netty#632]
Since com.sun.nio.sctp.SctpChannel is a InterruptibleChannel, OioEventLoop's thread.interrupt() usage (to switch read/write in single threaded event loop) closes the channel prematurely. 

java.nio.channels.ClosedByInterruptException
java.nio.channels.ClosedByInterruptException
    at java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:202)
    at sun.nio.ch.SctpChannelImpl.receive(SctpChannelImpl.java:772)
    at sun.nio.ch.SctpChannelImpl.receive(SctpChannelImpl.java:730)
    at io.netty.channel.socket.oio.OioSctpChannel.doReadMessages(OioSctpChannel.java:117)
    at io.netty.channel.socket.oio.AbstractOioMessageChannel$OioMessageUnsafe.read(AbstractOioMessageChannel.java:45)
    at io.netty.channel.socket.oio.OioEventLoop.run(OioEventLoop.java:81)
    at io.netty.channel.SingleThreadEventExecutor$1.run(SingleThreadEventExecutor.java:80)
    at java.lang.Thread.run(Thread.java:722)

So I did following change in OioEventLoop to get it working. This change was merged into the master  without review (https://github.com/netty/netty/pull/619) :-) .

OioEventLoop: before OioSctpChannel

```
protected void wakeup(boolean inEventLoop) {
    interruptThread();
}
```

OioEventLoop:  change after OioSctpChannel pull request

```
protected void wakeup(boolean inEventLoop) {
    if (!inEventLoop && isShutdown()) {
        interruptThread();
    }
}
```

Now problem is,  OioSctpChannel's write() does not work properly sometime because, the event loop thread is blocked on ch.unsafe().read(); foreever without processing task queue.

Can we come up with a separate EventLoop for  blocking InterruptibleChannels or is there any other way to solve this issue?

Please note that, I could not reproduce this issue with OioSocketChannel or even in OioSctpEchoClient/Server :)
